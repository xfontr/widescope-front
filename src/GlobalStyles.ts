import { createGlobalStyle } from "styled-components";

const GlobalStyles = createGlobalStyle`
  * {
    box-sizing: border-box;
  }

  body {
    margin: 0;
    min-height: 100vh;
  }

  ul {
    list-style: none;
    margin: 0;
    padding: 0;
  }

  button, input {
    font: inherit;
    border: none;
  }
`;

export default GlobalStyles;
