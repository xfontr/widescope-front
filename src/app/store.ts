import { configureStore, ThunkAction, Action } from "@reduxjs/toolkit";
import { userDataReducer } from "../store/slices/userData/userDataSlice";
import { userReducer } from "../store/slices/user/userSlice";

export const store = configureStore({
  reducer: {
    user: userReducer,
    userData: userDataReducer,
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
