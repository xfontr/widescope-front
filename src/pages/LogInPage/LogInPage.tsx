import SignForm from "../../components/SignForm/SignForm";

const LogInPage = (): JSX.Element => <SignForm isLogin={true} />;

export default LogInPage;
