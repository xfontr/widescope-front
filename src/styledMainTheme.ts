const styledMainTheme = {
  colors: {
    primary: "rgb(255, 255, 255)",
    primaryDarker: "rgb(238, 238, 238)",

    secondary: "rgb(25, 29, 35)",
    secondaryBrigther: "rgba(25, 29, 35, 0.7)",
  },

  shapes: {
    radiusSmall: "0.75rem",
    radiusBig: "1.5rem",

    shortShadow: "6px 6px 18px 6px rgba(0, 0, 0, 0.1)",
    longShadow: "10px 10px 30px 10px rgba(0, 0, 0, 0.1)",

    rectangularContainer: "0.8rem 1.5rem 0.8rem 1.5rem",
  },

  spacing: {
    gapSmall: "1rem",
    gapBig: "1.5rem",

    paddingSmall: "0.8rem",
    paddingBig: "1.5rem",
  },

  breakpoints: {
    small: "390px",
    big: "700pxpx",
    large: "1050px",
  },
};

export default styledMainTheme;
