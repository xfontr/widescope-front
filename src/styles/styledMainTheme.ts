const styledMainTheme = {
  colors: {
    primary: "rgb(255, 255, 255)",
    primaryDarker: "rgb(238, 238, 238)",

    secondary: "rgb(25, 29, 35)",
    secondaryBrigther: "rgba(25, 29, 35, 0.7)",

    error: "rgb(179, 120, 120)",
    success: "green",
  },

  shapes: {
    radiusSmall: "0.4rem",
    radiusBig: "1.5rem",

    shortShadow: "6px 6px 18px 6px rgba(0, 0, 0, 0.1)",
    longShadow: "10px 10px 30px 10px rgba(0, 0, 0, 0.1)",

    rectangularContainer: "0.7rem 1.6rem",
  },

  spacing: {
    gapSmall: "1.7rem",
    gapBig: "2.5rem",

    paddingSmall: "0.8rem",
    paddingBig: "1.8rem",
  },

  breakpoints: {
    small: "390px",
    big: "700px",
    large: "1250px",
  },
};

export default styledMainTheme;
