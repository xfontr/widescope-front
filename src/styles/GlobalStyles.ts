import { createGlobalStyle } from "styled-components";
import styledMainTheme from "./styledMainTheme";

const GlobalStyles = createGlobalStyle`
  * {
    box-sizing: border-box;
  }

  body {
    margin: 0;
    min-height: 100vh;
    background-color: ${styledMainTheme.colors.primaryDarker};
  }

  ul {
    list-style: none;
    margin: 0;
    padding: 0;
  }

  button, input {
    font: inherit;
    border: none;
  }
`;

export default GlobalStyles;
