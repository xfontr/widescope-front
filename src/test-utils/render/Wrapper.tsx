import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import { ThemeProvider } from "styled-components";
import { store } from "../../app/store";
import GlobalStyles from "../../styles/GlobalStyles";
import styledMainTheme from "../../styles/styledMainTheme";

interface WrapperProps {
  children: JSX.Element | JSX.Element[];
}

const Wrapper = ({ children }: WrapperProps): JSX.Element => {
  return (
    <ThemeProvider theme={styledMainTheme}>
      <Provider store={store}>
        <BrowserRouter>
          <GlobalStyles />
          {children}
        </BrowserRouter>
      </Provider>
    </ThemeProvider>
  );
};

export default Wrapper;
