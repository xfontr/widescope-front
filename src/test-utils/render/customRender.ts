import { render } from "@testing-library/react";
import { Wrapper } from "./Wrapper";

const customRender = (view: JSX.Element, ...options: any) =>
  render(view, { wrapper: Wrapper, ...options });

export * from "@testing-library/react";

export { customRender as render };
