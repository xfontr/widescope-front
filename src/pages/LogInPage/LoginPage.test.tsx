import { render, screen } from "../../test-utils/render/customRender";
import LogInPage from "./LogInPage";
import renderer from "react-test-renderer";
import { store } from "../../app/store";
import { Provider } from "react-redux";
import { ThemeProvider } from "styled-components";
import { BrowserRouter } from "react-router-dom";
import GlobalStyles from "../../styles/GlobalStyles";
import styledMainTheme from "../../styles/styledMainTheme";

describe("Given a LoginUpPage component", () => {
  describe("When instantiated", () => {
    test("Then it should match the snapshot", () => {
      const expectedPage = renderer.create(
        <Provider store={store}>
          <ThemeProvider theme={styledMainTheme}>
            <BrowserRouter>
              <GlobalStyles />
              <LogInPage />
            </BrowserRouter>
          </ThemeProvider>
        </Provider>
      );

      expect(expectedPage).toMatchSnapshot();
    });

    test("Then it should show a sign up form", () => {
      render(<LogInPage />);

      const signUpHeading = screen.getByRole("heading", {
        name: "Log in",
        level: 3,
      });

      expect(signUpHeading).toBeInTheDocument();
    });
  });
});
