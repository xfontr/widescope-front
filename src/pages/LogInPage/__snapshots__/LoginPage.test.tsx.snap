// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a LoginUpPage component When instantiated Then it should match the snapshot 1`] = `
Array [
  <h2
    className="page__title"
  >
    <span
      className="page__title--bold"
    >
      Log in
    </span>
  </h2>,
  <form
    className="sc-hKMtZM ctpeAE"
    data-testid="form"
    onSubmit={[Function]}
  >
    <header
      className="sc-eCYdqJ jizDxD"
    >
      <h3
        className="form__heading"
      >
        Log in
      </h3>
      <span
        className="form__cta-text"
      >
        Share your latests projects with the WideScope community.
      </span>
    </header>
    <div
      className="sc-jSMfEi icykpQ"
    >
      <label
        className="sc-gKXOVf jsDXqo"
        htmlFor="name"
      >
        Name
      </label>
      <input
        className="sc-iBkjds jsjeVM  form__input--user"
        id="name"
        name="name"
        onChange={[Function]}
        placeholder="John Doe"
        type="text"
        value=""
      />
    </div>
    <div
      className="sc-jSMfEi icykpQ"
    >
      <label
        className="sc-gKXOVf jsDXqo"
        htmlFor="password"
      >
        Password
      </label>
      <input
        className="sc-iBkjds jsjeVM  form__input--password"
        id="password"
        name="password"
        onChange={[Function]}
        type="password"
        value=""
      />
    </div>
    <footer
      className="sc-ftvSup kVrNfS"
    >
      <button
        className="sc-bczRLJ jZmJeh undefined button--default "
        onClick={[Function]}
        rel="noopener noreferrer"
        target="_blank"
        type="submit"
      >
        Log in
      </button>
      <span
        className="form__sign-up-cta"
      >
        Don't have an account?
        <a
          href="/sign-up"
          onClick={[Function]}
        >
          Sign up for free
        </a>
      </span>
    </footer>
  </form>,
]
`;
